sympy/matrices/expressions/matmul.py

```diff
@@ -3,7 +3,7 @@
 from sympy.core import Mul, Basic, sympify, Add
 from sympy.core.compatibility import range
 from sympy.functions import adjoint
-from sympy.matrices.expressions.transpose import transpose
+from sympy.functions.elementary.complexes import transpose
```

Comment:
That's what I call "experimental programming": If the function that's called doesn't work, let's hope another, similarly-named one from another module will.
Except it's just fixing symptoms unless the other function is truly the one that should have been called - which is extremely unlikely because we're inside `matrices`.
Also, I doubt it's even fixing symptoms. Try adding a test that transposes a 4x3 matrix; I'm pretty sure that this will fail.